src/libs/Search/searchParser.peggy

```diff
@@ -0,0 +1,104 @@
+// src/libs/Search/searchParser.peggy
+{
+  const defaultValues = {
+    "type": "expense",
+    "status": "all",
+    "sortBy": "date",
+    "sortOrder": "desc",
+    "offset": 0
+  };
+
+  function buildFilter(operator, left, right) {
+    return { operator, left, right };
+  }
+
+  function applyDefaults(filters) {
+    return {
+      ...defaultValues,
+      filters
+    };
+  }
+  
+  function updateDefaultValues(field, value) {
+    defaultValues[field] = value;
+  }
+
+  function isDefaultField(field) {
+    return defaultValues.hasOwnProperty(field);
+  }
+}
+
+query
+  = _ filters:filterList? _ { return applyDefaults(filters); }
+
+filterList
+  = head:filter tail:(logicalAnd filter)* {
+      const allFilters = [head, ...tail.map(([_, filter]) => filter)].filter(filter => filter !== null);
+      if (!allFilters.length) {
+      	return null;
+      }
+      return allFilters.reduce((result, filter) => buildFilter("and", result, filter));
+    }
+
+filter
+  = _ field:key? _ op:operator? _ value:identifier {
+      if (isDefaultField(field)) {
+        updateDefaultValues(field, value.trim());
+        return null;
+      }
+
+      if (!field && !op) {
+        return buildFilter('eq', 'freeText', value.trim());
+      }
+
+      const values = value.split(',');
+      return values.slice(1).reduce((acc, val) => buildFilter('or', acc, buildFilter('eq', field, val.trim())), buildFilter('eq', field, values[0]));
```

Comment:
I noticed that we weren't capturing the correct operators, so we need to change the grammar a bit:
```suggestion
      const values = value.split(',');
      const operatorValue = op ?? 'eq';
      return values.slice(1).reduce((acc, val) => buildFilter('or', acc, buildFilter(operatorValue, field, val.trim())), buildFilter(operatorValue, field, values[0]));

```